File: lib/models/san_pham.dart
import 'package:kingofshoes/models/bien_the_san_pham.dart';

class SanPham {
  final int id;
  final String ten;
  final double gia;
  final String moTa;
  final String idAnh;
  final List<BienTheSanPham> bienThe;

  SanPham(
      {required this.id,
      required this.ten,
      required this.gia,
      required this.idAnh,
      required this.moTa,
      required this.bienThe});

  factory SanPham.fromJson(Map<String, dynamic> json) {
    return SanPham(
      id: json['id'] ?? 0,
      ten: json['ten'] ?? '',
      gia: double.tryParse(json['gia'] ?? '0') ?? 0.0,
      moTa: json['mo_ta'] ?? '',
      idAnh: json['id_anh']['duong_dan_anh'] ?? '',
      bienThe: json['bien_the'] != null
          ? List<BienTheSanPham>.from(json['bien_the']
              .map((bienThe) => BienTheSanPham.fromJson(bienThe)))
          : [],
    );
  }
}

File: app/Models/SanPham.php
<?php
namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Session;
use App\Models\BienTheSanPham;
use App\Models\NhaCungCap;
class SanPham extends Model {
    use HasFactory;
      // Tắt timestamps
    public $timestamps = false;
    protected $table = 'san_pham';
    protected $fillable = ['ten', 'id', 'mo_ta', 'nhacungcap_id', 'loai_san_pham'];
    public function bienThe() {
      return $this->hasMany(BienTheSanPham::class, 'san_pham_id', 'id');
  }
}


